### PR TITLE
FIX: Render detailed_404 page from 403 responses

### PR DESCRIPTION
**Pull request by riking**
https://meta.discourse.org/t/detailled-404-for-category-page/171444/

This was previously broken by 59ef48c0b9d70a6639a545d1c96f3cc6f93ebf2c (#11425, #11424, https://meta.discourse.org/t/error-page-appears-repeatedly-after-you-try-accessing-a-private-page/170557/).

Centralize the logic into the exception controller, which avoids the problematic bug and makes it easy to add additional detailed 404 pages in the future.

:warning: This commit is a localization break. Translators should copy the `not_in_group.title_topic` key to the `not_in_group.title_category` key, changing "topic" to "category".

### CODE REVIEW

**Comment by riking** [comment]
Bug: History stack is broken when you click a restricted category link (back goes back 2 steps).

**Comment by riking** [comment]
The second commit is a slight improvement to the history stack mess caused by the exception page.

File: app/assets/javascripts/discourse/app/controllers/exception.js [hunk 1]
```diff
@@ -51,6 +52,10 @@ export default Controller.extend({
   isServer: gte("thrown.status", 500),
   isUnknown: none("isNetwork", "isServer"),
 
+  // Handling for the detailed_404 setting (which actually creates 403s)
+  errorHtml: alias("thrown.responseJSON.extras.html"),
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/controllers/exception.js:56]: I wonder if we need a CP for this. Might be simpler to omit and let the template use the `and` helper for `useErrorHtml` too.

File: config/locales/client.en.yml [hunk 2]
```diff
@@ -1667,13 +1667,15 @@ en:
...
         unknown: "Error"
+        no_data: "Error Not Found"
         not_found: "Page Not Found"
       desc:
         network: "Please check your connection."
         network_fixed: "Looks like it's back."
         server: "Error code: %{status}"
         forbidden: "You're not allowed to view that."
         not_found: "Oops, the application tried to load a URL that doesn't exist."
+        no_data: "Something went wrong, or you may have visited this page directly."
```
Comment (eviltrout) [inline config/locales/client.en.yml:1678]: I don't quite understand what "you many have visited this page directly" means, and I worry if I don't the end user will not! Is there something more actionable we can say?

Comment (riking) [reply config/locales/client.en.yml:1678]: Yeah, I wasn't too sure. This is what gets shown when you visit `/exception`, which was only generated in some intermediate code revisions (I decided against intentionally generating it for now).

Comment (eviltrout) [reply config/locales/client.en.yml:1678]: I'd prefer "Something went wrong" without the other part in that case.

File: app/assets/javascripts/discourse/app/controllers/exception.js (hunk 1 above)
Comment (riking) [reply app/assets/javascripts/discourse/app/controllers/exception.js:56]: I couldn't quite figure out what you were saying here, so I just moved the `and` into the template, eliminating the `useErrorHtml` property. I kept the deep nesting alias.

File: config/locales/client.en.yml (hunk 2 above)
Comment (riking) [reply config/locales/client.en.yml]: Done

**Review by eviltrout — APPROVED** [review]
Looks good! Please merge when you know you are around to respond to any regressions that might pop up.

**Pull request merged**